ingest.py

```diff
@@ -83,60 +83,89 @@ def load(self) -> List[Document]:
 
 def load_single_document(file_path: str) -> Document:
     ext = "." + file_path.rsplit(".", 1)[-1]
-    if ext in LOADER_MAPPING:
-        loader_class, loader_args = LOADER_MAPPING[ext]
-        loader = loader_class(file_path, **loader_args)
-        return loader.load()[0]
 
-    raise ValueError(f"Unsupported file extension '{ext}'")
+    loader_class, loader_args = LOADER_MAPPING[ext]
+    loader = loader_class(file_path, **loader_args)
+    return loader.load()[0]
 
 
-def load_documents(source_dir: str, ignored_files: List[str] = []) -> List[Document]:
+def load_documents(
+    source_dir: str,
+    ignored_files: Optional[List[str]] = None,
+    report_processed_files: bool = True,
+    report_skipped_files: bool = True,
+    report_ignored_files: bool = True,
+    use_process_bar: bool = True,
+) -> List[Document]:
+
     """
     Loads all documents from the source documents directory, ignoring specified files
     """
-    all_files = []
-    for ext in LOADER_MAPPING:
-        all_files.extend(
-            glob.glob(os.path.join(source_dir, f"**/*{ext}"), recursive=True)
-        )
-    filtered_files = [file_path for file_path in all_files if file_path not in ignored_files]
+    if ignored_files is None:
+        ignored_files : List[str] = []
+
+    filtered_files = []
+    extensions = tuple(LOADER_MAPPING.keys())
+
+    for root, _, files in os.walk(source_dir):
+        for file in files:
+            if file.endswith(extensions):
+                file_path = os.path.join(root, file)
+                if file_path not in ignored_files:
+                    filtered_files.append(file_path)
+                else:
+                    if report_ignored_files:
+                        print(f"Ignored '{file_path}' (ignore list)")
+            else:
+                if report_skipped_files:
+                    file_path = os.path.join(root, file)
+                    print(f"Skipping '{file_path}' - unmatched extension")
 
     with Pool(processes=os.cpu_count()) as pool:
         results = []
-        with tqdm(total=len(filtered_files), desc='Loading new documents', ncols=80) as pbar:
+        with (tqdm(total=len(filtered_files), desc='Loading new documents', ncols=80) if use_process_bar else None) as pbar:
```

Comment:
With `use_process_bar=False` this crashes. Maybe just let the progress bar always on? It doesn't seem worth the hassle to make it optional.
BTW, is `process` intentional or did you mean `progress`?

Reply:
"process bar" and "progress bar" both seem to be used in the wild, but here it should be "progress" which seems more popular -> renamed.

The crash is also fixed.